assets/Plausible.ts

```diff
@@ -1,77 +1,89 @@
-export const plausible = (name: string, options: { url?: string; props?: { [key: string]: string } } = {}) => {
-    const plausibleURL = env.isProd ? "https://p.chatanalytics.app" : "http://localhost:8000";
+const PLAUSIBLE_URL = "env" in window && env.isProd ? "https://p.chatanalytics.app" : "http://localhost:8000";
+const ALLOWED_PATHS = ["", "/demo"];
 
-    // Avoid leaking personal information
-    // it happened in the past, we don't know why
-    // so we'll try to a void it replace any URL
-    // we don't recognize
-    let url = options.url || window.location.href;
-    if (!url.startsWith("https://chatanalytics.app")) {
-        url = "https://chatanalytics.app/report";
+export const plausible = (name: "pageview" | string, props?: { [key: string]: string }) => {
+    // We want the url where:
+    // - the hostname is always "chatanalytics.app"
+    //   - otherwise fallback to "/report"
+    // - the path is either "", "/", "/demo" or "/report"
+    //   - otherwise fallback to "/report"
+    // - all search params are kept
+
+    let pathname: string;
+    if (window.location.hostname === "chatanalytics.app") {
+        pathname = window.location.pathname;
+        if (pathname.endsWith("/")) {
+            pathname = pathname.slice(0, -1); // make sure the path doesn't end with a slash
+        }
+        if (!ALLOWED_PATHS.includes(pathname)) {
+            pathname = "/report";
+        }
+    } else {
+        pathname = "/report";
```

Comment:
I think this it a bit harder to read to what was before

Reply:
How so? I think it makes it clearer that the pathname is only being changed if the expression matches. Ternary expressions suggest it will be changed either way.